documents/processor.go

```diff
@@ -139,17 +169,23 @@ func (dp defaultProcessor) AnchorDocument(ctx context.Context, model Model) erro
 		return errors.New("failed to get document root: %v", err)
 	}
 
-	anchorID, err := anchors.ToAnchorID(model.CurrentVersion())
+	anchorID, err := anchors.ToAnchorID(model.CurrentVersionPreimage())
```

Comment:
Shouldn't it the pre-image of the previous document version?

Reply:
no we generate the new document with correct versions so this is the way it is now